tests/components/normal-components/transistor.test.tsx

```diff
@@ -88,3 +89,57 @@ it("should render a PNP transistor with 270° rotation", async () => {
   circuit.render()
   expect(circuit).toMatchSchematicSnapshot(import.meta.path + ".pnp.270")
 })
+it("should have base, emitter, and collector port mappings", async () => {
+  const { circuit } = getTestFixture()
+  circuit.add(
+    <board width="10mm" height="10mm">
+      <transistor name="Q3" type="npn" schRotation={0} />
+    </board>,
+  )
+  circuit.render()
+  const transistorInstance = circuit.selectOne("Transistor") as Transistor
+  expect(transistorInstance).toBeDefined()
+  expect(transistorInstance.base).toBeDefined()
+  expect(transistorInstance.emitter).toBeDefined()
+  expect(transistorInstance.collector).toBeDefined()
+})
+
+it("should initialize base, emitter, and collector ports correctly for an NPN transistor", async () => {
+  const { circuit } = getTestFixture()
+  circuit.add(
+    <board width="10mm" height="10mm">
+      <transistor name="Q1" type="npn" schRotation={0} />
+    </board>,
+  )
+  circuit.render()
+
+  const components = circuit.db.pcb_component.list()
+  expect(components).toHaveLength(1)
+  const component = components[0]
+
+  const ports = circuit.db.pcb_port.list()
+
+  const base = ports.find((port) => port.source_port_id === "b")
```

Comment:
Why the `source_port_id` just letter? It shouldn't be the case it should follow the standard format!